src/media-segment-request.js

```diff
@@ -350,39 +359,45 @@ const handleSegmentBytes = ({
 }) => {
   const bytesAsUint8Array = new Uint8Array(bytes);
 
-  segment.isFmp4 =
-    // only set the property the first time we see some bytes so that partial appends
-    // don't try to check every section of bytes (since the check should only consider the
-    // first bytes in the segment)
-    typeof segment.isFmp4 === 'boolean' ? segment.isFmp4 : isLikelyFmp4Data(bytesAsUint8Array);
-
-  if (segment.isFmp4) {
+  if (isLikelyFmp4Data(bytes)) {
+    segment.isFmp4 = true;
+    const videoTrack = segment.map.tracks.video;
+    let audioTrack = segment.map.tracks.audio;
 
     // since we don't support appending fmp4 data on progress, we know we have the full
     // segment here
-    const startTime = probeMp4StartTime(bytesAsUint8Array, segment.map.bytes);
-
-    // we don't parse fmp4 (yet), so we can't provide any track info, however, the
-    // track info callback should still be called before the dataFn is called
-    trackInfoFn(segment, null);
-    // use null for the media type since we don't technically know whether it's audio or
-    // video
+    trackInfoFn(segment, {
+      hasVideo: videoTrack ? true : false,
+      hasAudio: audioTrack ? true : false,
+      audioCodec: (audioTrack || {}).codec,
```

Comment:
is it bad for the codec to be set to {}.codec? In other words, `undefined`? I don't think we should allow that

Reply:
I'm not sure there is much of a difference between not having the `audioCodec` key in the track info object and having it set to `undefined`.

Reply:
It may get confusing, especially if we accidentally end up trying to create a sourceBuffer with `undefined` codec

Reply:
In that case though `hasVideo` will either be `false` or we will use the `playlist` or `default` codec.

Reply:
ah, yeah I guess that would make sense for debugging too